### PR TITLE
fix: SDO date of issue always set to today (FPLA-1791)

### PR DESCRIPTION
**Pull request by joez-f**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1791

### Change description ###

SDO date of issue is no longer overwritten with today's date when re-visiting the event (pre-populate with previously entered date instead)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersController.java [hunk 1]
```diff
@@ -79,8 +81,14 @@
...
     public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        caseDetails.getData().put("dateOfIssue", time.now().toLocalDate());
+        if (isEmpty(caseData.getStandardDirectionOrder().getDateOfIssue())) {
+            caseDetails.getData().put("dateOfIssue", time.now().toLocalDate());
+        } else {
+            caseDetails.getData().put("dateOfIssue",
+                parseLocalDateFromStringUsingFormat(caseData.getStandardDirectionOrder().getDateOfIssue(), DATE));
```
Comment (swalker125) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersController.java:90]: let's have the put code once and then just set a variable to the date in the if statements

Comment (joez-f) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersController.java:90]: 👍

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersController.java [hunk 2]
```diff
@@ -77,10 +79,17 @@
     private static final String JUDGE_AND_LEGAL_ADVISOR_KEY = "judgeAndLegalAdvisor";
 
     @PostMapping("/about-to-start")
-    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
-        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+        LocalDate issuedDate = time.now().toLocalDate();
```
Comment (akomisarek) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersController.java:85]: Not sure why  naming is not consistent -  issuedDate vs dateOfIssue?

Comment (joez-f) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersController.java]: fixed 👍

**Pull request merged**